### PR TITLE
refactor(octicons_react): update package deps and tests

### PR DESCRIPTION
**Pull request by joshblack**
This PR introduces some prep work before https://github.com/primer/octicons/pull/837 moves forward. In particular, it adds in some Public API and tree-shaking tests to protect against any regressions if an export no longer exists or the bundle can no longer be tree-shaken.

**What's changed**

- Add PublicAPI.test.js which snapshots the exports from the entrypoint
- Add tree-shaking.test.js which verifies that unused exports can be tree-shaken with no bundle impact
- Update babel deps
- Update rollup deps
- Update Next.js preview
  - Update Next.js to v12
  - Update preview to use the new icon format
- Update octicon test to no longer emit `console.warn` for the deprecated `Octicon` export

### CODE REVIEW

File: lib/octicons_react/__tests__/PublicAPI.test.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const Octicons = require('@primer/octicons-react')
+
+describe('@primer/octicons-react', () => {
+  it('should not update exports without a semver change', () => {
+    expect(Object.keys(Octicons).sort()).toMatchSnapshot()
```
Comment (colebemis) [inline lib/octicons_react/__tests__/PublicAPI.test.js:5]: ❤️

File: lib/octicons_react/__tests__/tree-shaking.test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        throw new Error(message)
+      }
+    }
+  })
+  const {output} = await bundle.generate({
+    format: 'esm'
+  })
+
+  for (const {code} of output) {
+    expect(code.trim()).toBe('')
```
Comment (colebemis) [inline lib/octicons_react/__tests__/tree-shaking.test.js:28]: If I'm reading this correctly, this test asserts that unused imports don't get bundled. Do we also need to check that importing and use a single icon doesn't bring all the icons with it?

**Review by colebemis — APPROVED** [review]

File: lib/octicons_react/__tests__/PublicAPI.test.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const Octicons = require('@primer/octicons-react')
```
[the remark this answers is not in the document]
Comment (joshblack) [reply lib/octicons_react/__tests__/PublicAPI.test.js:1]: @colebemis honestly not really haha. I guess Public API felt like a proper noun? I'll update it over to `public-api.test.js` though 👍

File: lib/octicons_react/__tests__/tree-shaking.test.js (hunk 2 above)
Comment (joshblack) [reply lib/octicons_react/__tests__/tree-shaking.test.js:28]: @colebemis would be great to add, I'm just not sure what the assertion would look like 🤔 let me know if you have any ideas!

**Pull request merged**